airflow/providers/amazon/aws/operators/sagemaker.py

```diff
@@ -158,6 +159,14 @@ def hook(self):
         """Return SageMakerHook."""
         return SageMakerHook(aws_conn_id=self.aws_conn_id)
 
+    @staticmethod
+    def path_to_s3_dataset(path):
```

Comment:
```suggestion
    def path_to_s3_dataset(path) -> Dataset:
```
(and it needs to be added to the imports under `if TYPE_CHECKING:` as well)

Reply:
Done.

Reply:
hmm, it looks like you forgot to push after that maybe ?

Reply:
@vandonr-amz you're nearly right - I've pushed but to wrong upstream. Neverthless it's here: https://github.com/apache/airflow/pull/33114